packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -90,35 +99,168 @@ class BottomNavigationBar extends StatefulWidget {
   /// The index into [items] of the current active item.
   final int currentIndex;
 
-  /// Defines the layout and behavior of a [BottomNavigationBar].
+  /// Defines the layout and behavior of a [BottomNavigationBar].  See
+  /// documentation for [BottomNavigationBarType] for information on the meaning
+  /// of different types.
   final BottomNavigationBarType type;
 
   /// The color of the selected item when bottom navigation bar is
-  /// [BottomNavigationBarType.fixed].
+  /// [BottomNavigationBarType.fixed].  [fixedColor] must be null if the [type]
+  /// is not [BottomNavigationBarType.fixed].
   final Color fixedColor;
 
   /// The size of all of the [BottomNavigationBarItem] icons.
   ///
   /// This value is used to to configure the [IconTheme] for the navigation
-  /// bar. When a [BottomNavigationBarItem.icon] widget is not an [Icon] the widget
-  /// should configure itself to match the icon theme's size and color.
+  /// bar. When a [BottomNavigationBarItem.icon] widget is not an [Icon] the
+  /// widget should configure itself to match the icon theme's size and color.
   final double iconSize;
 
   @override
   _BottomNavigationBarState createState() => new _BottomNavigationBarState();
 }
 
+// This represents a single tile in the bottom navigation bar.  It is intended
+// to go into a flex container.
+class _BottomNavigationTile extends StatelessWidget {
+  _BottomNavigationTile(
+    this.type,
+    this.item,
+    this.onTap,
+    this.animation,
+    this.iconSize,
+    {this.colorTween, this.flex}
```

Comment:
format this as:
```dart
  _ClassThingy(
￼    this.foo,
    this.bar, {
    this.baz,
    this.quux,
  });
```

Reply:
OK, done, but why aren't `this.baz` and `this.quux` indented two additional spaces?  That really wreaks havoc with most auto-indents (and my brain: it's easy to miss that they are optional).

Reply:
It's mostly because otherwise it becomes hard to justify only using two indents when you only have named arguments. :-)

I agree it's not the most obvious style.